src/chrome/logging/methodsCalledLogger.ts

```diff
@@ -59,38 +58,48 @@ export class MethodsCalledLogger<T extends object> {
                                 args[1] = listenerPossiblyWrapped;
                             }
 
+                            this.logCallStart(propertyKey, args, callId);
                             const result = originalPropertyValue.apply(target, args);
                             if (!result || !result.then) {
-                                this.logCall(propertyKey, Synchronicity.Sync, args, Outcome.Succesful, result);
-                                let resultPossiblyWrapped = result;
-                                this._configuration.decideWhetherToWrapMethodResult(propertyKey, args, result, name => resultPossiblyWrapped = new MethodsCalledLogger(this._configuration, result, name).wrapped());
-                                return resultPossiblyWrapped;
+                                this.logCall(propertyKey, Synchronicity.Sync, args, Outcome.Succesful, result, callId);
+                                if (result === target) {
+                                    return receiver;
+                                } else {
+                                    let resultPossiblyWrapped = result;
+                                    this._configuration.decideWhetherToWrapMethodResult(propertyKey, args, result, name => resultPossiblyWrapped = new MethodsCalledLogger(this._configuration, result, name).wrapped());
+                                    return resultPossiblyWrapped;
+                                }
                             } else {
-                                const callId = this.generateCallId();
-                                this.logCallStart(propertyKey, args, callId);
+                                this.logSyncPartFinished(propertyKey, args, callId);
                                 return result.then((promiseResult: unknown) => {
                                     this.logCall(propertyKey, Synchronicity.Async, args, Outcome.Succesful, promiseResult, callId);
-                                    let resultPossiblyWrapped = promiseResult;
-                                    this._configuration.decideWhetherToWrapMethodResult(propertyKey, args, promiseResult, name => resultPossiblyWrapped = new MethodsCalledLogger(this._configuration, <object>promiseResult, name).wrapped());
-                                    return resultPossiblyWrapped;
+                                    if (promiseResult === target) {
+                                        return receiver;
+                                    } else {
+                                        let resultPossiblyWrapped = promiseResult;
+                                        this._configuration.decideWhetherToWrapMethodResult(propertyKey, args, promiseResult, name => resultPossiblyWrapped = new MethodsCalledLogger(this._configuration, <object>promiseResult, name).wrapped());
```

Comment:
Two notes on the wrapping logic here.

1. This seems like a really complicated way to wrap the name for emitters (create a variable, pass it through a closure, then reassign/return it), and we repeat this logic several times. Could we just have the method return a string and assign it directly rather than passing the variable through a closure?

2. Is there really a need to make this logic so abstract/configurable? I only see one implementation of `IMethodsCalledLoggerConfiguration` and the method `decideWhetherToWrapEventEmitterListener` seems extremely specific to this one use case. Is there ever a case where we'd override this behavior, or really need another implementation of that interface?

Reply:
1. Do you mean returning the name for the object instead of assigning the full MethodsCalledLogger?
2. In the test we manually wrap the Wizard classes in the MethodsLogger to log the actions. A better approach than doing that manually, could be to automatically detect when a Wizard is returned, and wrap it with a proxy automatically. If we do that, we'd use this method/interface

Reply:
In response to (1) Why not just let it return the object, wrapped or not, with any name changes the implementation feels like adding? `decideWhetherToWrapMethodResult` is already taking the `result` object as a param, why can't he just return the result unaltered if he chose not to wrap it, and return the wrapped object (the passed wrapping function would return the wrapped object rather than void). In addition to that, it looks like the wrapping function is mostly the same in all 3 places. We already have `wrapWithMethodLogger` which looks like it could do what we need for that. In which case, if we're passing (essentially) the same function for every invocation, why pass it? Just use `wrapWithMethodLogger`, or maybe pass it in the constructor to `MethodsCalledLoggerConfiguration` if we think that would ever change.

It all seems like a bit of a moot point anyway, since the only (current) implementation of `decideWhetherToWrapMethodResult` is a no-op, and I tried setting a breakpoint on line 69 and did just about everything I could in the debug adapter and couldn't get inside that block.

I'll approve and let you decide what you want to do with it, but It just seems like a lot of really confusing code that isn't actually doing anything \*right now\*, for the promise of _maybe_ being able to use it for something pretty minor in the future, and we've already spent way too much time on it. YAGNI.

Reply:
I think "return the object, wrapped or not" is a great idea! I'll change the code...

P.S.: decideWhetherToWrapMethodResult is overwritten/used in vscode-chrome-debug\test\int\utils\logging.ts